src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -343,13 +334,13 @@ private void init(long recordOffset)
     {
       logInfo("init pipe: {}", pipeName);
       // wait for sinkConnector to start
-      createTableAndStage();
+      SnowflakeTelemetryPipeCreation pipeCreation = createTableAndStage();
```

Comment:
Where is event reported ?

Reply:
In line 386 of this file

Reply:
It's very hard to understand the life-cycle of this record. i.e. where it is created where it is updated and where it is used.

Reply:
It would be much more understandable if the record was created and then "used" in the same function/block

Reply:
pipeCreation collects information in multiple stages (createTableAndStage, recover, startCleaner) of the pipe starting process.  I think I can move uploading pipeCreation telemetry from inside startCleaner to outside this function. So that the life cycle of this object is obvious in this function. We create the object, then update its value in multiple functions then upload it.